src/qibo/tomography/gate_set_tomography.py

```diff
@@ -278,7 +292,15 @@ def GST(
 
     for gate in gate_set:
         if gate is not None:
-            init_args = signature(gate).parameters
+            init_args = signature(gate[0]).parameters
+            params = gate[1]
+
+            angle_names = [name for name in init_args if name in {"theta", "phi"}]
```

Comment:
```suggestion
            angle_names = [name for name in init_args if name in {"theta", "phi", "lam"}]
```

This should cover any case, including cases like `gates.U2` and `gates.U3`.